### PR TITLE
Improve color provider

### LINKED ISSUE #104: Improve color provider
see: https://github.com/davidmerfield/randomColor

### PR DESCRIPTION
**Pull request by malefice**
### What does this changes

This will add a new color provider method `color()` that allows specifying a hue, a luminosity, and a color format to generate an matching color.

```
>>> from faker import Faker
>>> fake = Faker()

# Create random blue color in RGB format
>>> fake.color(hue='blue', color_format='rgb')
'rgb(5, 56, 175)'

# Create random bright red color in HSV format
>>> fake.color(hue='red', luminosity='bright', color_format='hsv')
'hsv(345, 96, 90)'

# Create random light pink color in HSL format
>>> fake.color(hue='pink', luminosity='light', color_format='hsl')
'hsl(294, 93, 71)'

# Create random color from yellow to green spectrum in hex format
>>> fake.color(hue=[50, 120], luminosity='dark', color_format='hex')
'#3f7c02'
``` 

### What was wrong

There is currently no support for such features.

### Notes

Personally, I prefer to return tuples of HSV, HSL, and RGB values instead of the `color_format(X, Y, Z)` string notation or any string for that matter, so that there is no need for users to parse the string for the values. I followed the current way `rgb_color()` and `rgb_css_color()` do this, but I am not a fan of those methods.

I also think that it would be better to provide more convenient methods and update old ones, so users will not need to remember the different `color_format` values while still being able to just specify 'blue' and 'dark' to get a color of that type.

Fixes #104

### CODE REVIEW

File: faker/providers/color/color.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+        if generator:
+            self.random = generator.random
+        else:
+            self.seed = seed if seed else random.randint(0, sys.maxsize)
+            self.random = random.Random(self.seed)
+
+        for color_name, color_attrs in self.colormap.items():
+            lower_bounds = color_attrs['lower_bounds']
+            s_min = lower_bounds[0][0]
+            s_max = lower_bounds[len(lower_bounds) - 1][0]
```
Comment (fcurella) [inline faker/providers/color/color.py]: isn't this equivalent to `s_max = lower_bounds[-1][0]`?

File: faker/providers/color/color.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+    def get_color_info(self, hue):
+        # Maps red colors to make picking hue easier
+        if 334 <= hue <= 360:
+            hue -= 360
+
+        for color_name, color in self.colormap.items():
+            if color['hue_range'][0] <= hue <= color['hue_range'][1]:
+                return self.colormap[color_name]
+        else:
+            raise ValueError('Value of hue is invalid.')
```
Comment (fcurella) [inline faker/providers/color/color.py]: can we also show the value itself in the exception message? Something like ``'Value of hue `%s` is invalid.' % hue``

File: faker/providers/color/color.py [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+
+        elif isinstance(color_input, six.string_types) and color_input in self.colormap:
+            return self.colormap[color_input]['hue_range']
+
+        try:
+            v1, v2 = color_input
+            v1 = int(v1)
+            v2 = int(v2)
+        except (ValueError, TypeError):
+            return [0, 360]
```
Comment (fcurella) [inline faker/providers/color/color.py]: I'm not sure I'd want it to silently fallback to defaults. I'd rather have it throw an exception if the user is passing invalid values.

Comment (malefice) [reply faker/providers/color/color.py]: Alright, I will change this behavior.

File: faker/providers/color/color.py (hunk 2 above)
Comment (malefice) [reply faker/providers/color/color.py]: This should only happen if something is wrong with `COLOR_MAP` itself, but sure.

File: faker/providers/color/color.py (hunk 1 above)
Comment (malefice) [reply faker/providers/color/color.py]: You are right. I wonder why the author of `randomcolor-py` did not just use `[-1]` for this one. I will fix this, and I am sorry I did not notice.

**Comment by malefice** [comment]
The flake8 errors are related to PR #1058.

**Comment by fcurella** [comment]
Looks Good! Thank you so much!

**Pull request merged**